app/Providers/AppServiceProvider.php

```diff
@@ -44,6 +47,16 @@ public function boot()
      */
     public function register()
     {
-        //
+        if ($this->app->environment() == 'testing') {
+            $this->app->singleton(Quote::class, QuoteFake::class);
+        } else {
+            $this->app->singleton(Quote::class, function(){
```

Comment:
Otherwise, use the real API.

You'll see here that the QuoteAPI implementation is wrapped in the the QuoteCache decorator. This makes it easy to add caching. If we want to remove it, we simple remove QuoteCache and return the QuoteAPI as is.

Eg. 
```
$this->app->singleton(Quote::class, function(){
    return new QuoteAPI(new Client());
});